cg/clients/chanjo2/api.py

```diff
@@ -0,0 +1,40 @@
+"""Coverage analysis API for human clinical sequencing data."""
+
+import logging
+from typing import Any
+
+import requests
+
+from cg.clients.chanjo2.models import CoverageData, CoverageRequest
+
+LOG = logging.getLogger(__name__)
+
+
+class Chanjo2APIClient:
+    """
+    Chanjo2 API to communicate with a d4tools software in order to return coverage
+    over genomic intervals.
+    """
+
+    def __init__(self, config: dict[str, str]):
+        self.headers = {
+            "Content-Type": "application/json",
+            "accept": "application/json",
+        }
+        self.host = config["chanjo2"]["host"]
+
+    def get_coverage(self, coverage_request: CoverageRequest) -> CoverageData | None:
+        """Send a POST request to the coverage endpoint to retrieve gene coverage summary data."""
+        endpoint: str = f"{self.host}/coverage/d4/genes/summary"
+        post_data: dict[str, Any] = coverage_request.model_dump()
+        try:
+            response = requests.post(url=endpoint, headers=self.headers, json=post_data)
+            response.raise_for_status()
+            response_content: dict[str, Any] = response.json().values()
+            if not response_content:
+                LOG.error("The POST get coverage response is empty")
+                return None
```

Comment:
Yes, also avoid returning None when something weird happens. This should be an exception and it would be thrown when trying to parse the data into the Pydantic model.

Reply:
The problem is that we can't allow this method to raise an error since we need to be able to force the generation of the delivery report even if the request fails. But I will move this logic to the Analysis API so that it is independent of the client API